shin-core/src/layout/layouter.rs

```diff
@@ -119,14 +120,24 @@ struct Layouter<'a> {
     pending_chars: Vec<LayoutedChar>,
     position: Vector2<f32>,
     time: Ticks,
+    metrics: MessageMetrics,
 }
 
 impl<'a> Layouter<'a> {
-    fn on_char(&mut self, c: char) {
+    fn on_char(&mut self, c: char, character_name: bool) {
```

Comment:
How about an enum `LayoutingMode` or something, that would include items `CharacterName` and `MessageText`, but also `GenericText` (or simply `Text`), and `LogText`, possibly `LogCharacterName`, etc.?

Reply:
This... Would be fine and should go into `LayoutParams` (without the `CharacterName` variant, as it's for the whole message)

I think I would be fine with `is_character_name` (or a better name) to be a part of `Layouter` struct if the logic in the message log and the normal message would match (0.9x font)

OR, even better, you can just follow what the game already does and just set font size to 0.9x when entering character name and to 1.0x when leaving (outsize of the `Layouter`). This is how why it ignores the size changes in the header btw

Reply:
Unfortunately the font size logic doesn't match, the character name and the main text are the same size in the message log:
![image](https://user-images.githubusercontent.com/102922815/211163791-52bfdfa1-05ef-4fe9-9095-b4c1c7557ba9.png)

But there is still unique behaviour in the message log: the character names are automatically surrounded by 【】. It might make sense to have `is_character_name` part of `Layouter`, and then depending on the `LayoutingMode`, either change the font size or add those brackets. In any case, the `Layouter` needs to be aware of whether we are in a character name or not, because it needs to know that it should put the chars somewhere else other than the main message box (unless this is done in some post processing step?)

Reply:
:thinking:
You can do this in post-processing if you somehow split the lines after the layouting (the orig engine does that, but I don't think it's used for anything

I kinda like the fact tat the post-processing would create less convoluted code in the `Layouter` (which is already pretty much a mess), but, on the other hand, it would make the system as a whole harder to understand...

I still think it would be nice to keep the `Layouter` implementation independent of the frontend, so I like the postprocessing route more rn

Reply:
Alright, I pushed an update that restructures a bunch of stuff, such that the character name positioning logic is now in `Message` rather than `Layouter`